### PR TITLE
Add release workflow for Python

### LINKED ISSUE #651: Release workflow for python layer
Not a bug, but a CI request. Similar to #646 but for Python.

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #651

### CODE REVIEW

File: .github/workflows/release-layer-python.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: "Release Python Lambda Layer"
+
+on:
+  # (Using tag push instead of release to allow filtering by tag prefix.)
+  push:
+    tags:
+      - layer-pythonjs/**
```
Comment (tylerbenson) [inline .github/workflows/release-layer-python.yml]: pythonjs?

File: .github/workflows/release-layer-python.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          path: python/packages/layer/build/opentelemetry-python-layer.zip
+
+  publish-layer:
+    uses: ./.github/workflows/layer-publish.yml
+    needs: build-layer
+    strategy:
+      matrix:
+        aws_region: 
+          - us-west-1
+          - us-west-2
```
Comment (tylerbenson) [inline .github/workflows/release-layer-python.yml:68]: probably want the other regions listed out in comments as a reminder for future official release.

Comment (ocelotl) [reply .github/workflows/release-layer-python.yml:68]: Fixed

File: .github/workflows/release-layer-python.yml (hunk 1 above)
Comment (ocelotl) [reply .github/workflows/release-layer-python.yml]: Fixed

**Review by tylerbenson — APPROVED** [review]

File: .github/workflows/release-layer-python.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'
+
+      - name: Configure aws
+        run: aws configure set region us-west-1
+
```
Comment (tylerbenson) [inline .github/workflows/release-layer-python.yml]:
```suggestion
```

I don't think this is needed anymore.

Comment (ocelotl) [reply .github/workflows/release-layer-python.yml]: Not sure about that, I removed that step and the workflow [failed](https://github.com/ocelotl/opentelemetry-lambda/actions/runs/4964002577/jobs/8883739381#step:4:18).

Comment (tylerbenson) [reply .github/workflows/release-layer-python.yml]: Why does it still need AWS access to build? I thought it was just doing a zip now.

Comment (ocelotl) [reply .github/workflows/release-layer-python.yml]: Ok, I refactored the script to only require `region` if it is publishing the layer by itself.

**Review by codeboten — APPROVED** [review]

**Pull request merged**